src/handlers/ResponseHandlers.ts

```diff
@@ -27,18 +28,49 @@ const successBlob: ResponseHandler = {
 const error: ResponseHandler = {
     canProcess: () => true,
     process: async <T>(response: Response): Promise<T> => {
-        let responseBody: any = await response.text();
+        const responseJson = await response.json();
         const platformError = new CoveoPlatformClientError();
+        Object.assign(platformError, responseJson);
+        platformError.xRequestId = response.headers.get('X-Request-ID') ?? 'unknown';
+        platformError.title = responseJson.title ?? getErrorTypeFromAliases(responseJson) ?? 'unknown';
+        platformError.detail = responseJson.detail ?? getErrorDetailFromAliases(responseJson) ?? 'unknown';
         platformError.status = response.status;
+        throw platformError;
+    },
+};
+
+const blockedByWAFHandler: ResponseHandler = {
+    canProcess: (response) => response.headers.get('x-coveo-waf-action') === 'block',
+    process: async (response) => {
+        const platformError = new CoveoPlatformClientError();
         platformError.xRequestId = response.headers.get('X-Request-ID') ?? 'unknown';
-        try {
-            responseBody = JSON.parse(responseBody);
-            Object.assign(platformError, responseBody);
-            platformError.title = responseBody.title ?? getErrorTypeFromAliases(responseBody) ?? 'unknown';
-            platformError.detail = responseBody.detail ?? getErrorDetailFromAliases(responseBody) ?? 'unknown';
-        } catch {
-            throw platformError;
-        }
+        platformError.title = 'Blocked for security reason'; // TODO: technical writer
+        platformError.detail = 'The web application firewall deemed that request to be dangerous so it was blocked'; // TODO: technical writer
+        platformError.status = response.status;
+        throw platformError;
+    },
+};
+
+const badGatewayHandler: ResponseHandler = {
+    canProcess: (response) => response.status === 502,
+    process: async (response) => {
+        const platformError = new CoveoPlatformClientError();
+        platformError.xRequestId = response.headers.get('X-Request-ID') ?? 'unknown';
+        platformError.title = 'The endpoint is not reachable ...'; // TODO: technical writer
+        platformError.detail = 'Probably because bad gateway'; // TODO: technical writer
+        platformError.status = response.status;
+        throw platformError;
+    },
+};
+
+const genericHTMLErrorHandler: ResponseHandler = {
+    canProcess: (response) => response.headers.get('content-type') === 'text/html',
+    process: async (response) => {
+        const platformError = new CoveoPlatformClientError();
+        platformError.xRequestId = response.headers.get('X-Request-ID') ?? 'unknown';
+        platformError.title = 'There is some issue with the endpoint and we cant parse the error'; // TODO: technical writer
+        platformError.detail = '...'; // TODO: technical writer
```

Comment:
In the detail here, I wonder if we couldn't just pass set the html response parsed as a string.

Reply:
As @JTangCoveo pointed out, could there be cases where the HTML string would be too long and wouldn't fit in the error prompt?

Reply:
The HTML response might be too long to display. The prompt might look something like this `<!DOCTYPE html><html lang="en"><head><meta charset="utf-8" /><title>test</title></head><body>test</body></html>`.

Reply:
@gdostie Let me know if you think we should include the HTML string in the detail @rtherien will need to know if he needs to write this part or not :)

Reply:
From a platform-client side of things, it makes sense to have an error detail property that is long, even a full html document is fine IMHO. Now, on the admin-ui side, you are right that we won't want to display HTML inside an error notification. I suggest that we adjust the admin-ui code for this use case instead of removing valuable information from the error object itself.

Reply:
Works for me 👍 .I will include the HTML string as the detail. And handle it from the adui side.